### PR TITLE
Pylint issues

### PR DESCRIPTION
**Pull request by marksmayo**
Tidied up .pylintrc file, removed unnecessary ignores and fixed a bunch of simple ones.

### CODE REVIEW

**Comment by wRAR** [comment]
Can you please look into the test failures and new flake8 messages?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (wRAR) [inline .vscode/settings.json:1]: I don't think this file should be committed

File: tests/test_request_cb_kwargs.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def parse_general(self, response, **kwargs):
             self.checks.append(kwargs['callback'] == 'some_callback')
             self.crawler.stats.inc_value('boolean_checks', 3)
         elif response.url.endswith('/general_without'):
-            self.checks.append(kwargs == {})
```
Comment (wRAR) [inline tests/test_request_cb_kwargs.py:94]: strictly speaking these conditions are different

**Comment by wRAR** [comment]
I think the test failure is because of the `unidiomatic-typecheck` "fix" (as the new code behaves differently in the case of a subclass).

File: tests/CrawlerRunner/ip_address.py [hunk 3]
```diff
@@ -30,9 +30,10 @@ def start_requests(self):
 
     def parse(self, response):
         netloc = urlparse(response.url).netloc
-        self.logger.info("Host: %s" % netloc.split(":")[0])
-        self.logger.info("Type: %s" % type(response.ip_address))
-        self.logger.info("IP address: %s" % response.ip_address)
+        a = netloc.split(":")[0]
```
[the remark this answers is not in the document]
Comment (felipeboffnunes) [reply tests/CrawlerRunner/ip_address.py:33]:
```suggestion
        host = netloc.split(":")[0]
```

File: tests/test_request_cb_kwargs.py [hunk 4]
```diff
@@ -91,7 +91,7 @@ def parse_general(self, response, **kwargs):
...
+            self.checks.append(not kwargs)
```
Comment (felipeboffnunes) [inline tests/test_request_cb_kwargs.py:94]:
```suggestion
            self.checks.append(kwargs == {})
```

File: scrapy/http/request/__init__.py [hunk 5]
```diff
@@ -187,7 +187,7 @@ def to_dict(self, *, spider: Optional["scrapy.Spider"] = None) -> dict:
         }
         for attr in self.attributes:
             d.setdefault(attr, getattr(self, attr))
-        if type(self) is not Request:
+        if not isinstance(self, Request):
```
Comment (felipeboffnunes) [inline scrapy/http/request/__init__.py:190]: Is this the one breaking the tests? to_dict fails on the pipeline and also isinstance caters to inheritance, whilst type does not. [https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance](Difference type and isinstance)

```suggestion
        if type(self) is not Request:
```

**Comment by Gallaecio** [comment]
@marksmayo Do you think you will find the time to work on addressing feedback here? Otherwise, @emarondan how would you feel about taking over and finishing this change yourself?

**Comment by felipeboffnunes** [comment]
I put some suggestions that could fix the current errors. We will still need to sync it with the Black PR #5734

File: scrapy/http/request/__init__.py (hunk 5 above)
Comment (felipeboffnunes) [inline scrapy/http/request/__init__.py:190]: `E   AssertionError: <class 'scrapy.http.request.form.FormRequest'> != <class 'scrapy.http.request.Request'>`

Is this the one breaking the tests? `test_request_dict` fails on the pipeline and also `isinstance` caters to inheritance, whilst `type` does not. 
[Difference type and isinstance](https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance)

```suggestion
        if type(self) is not Request:
```

File: tests/CrawlerRunner/ip_address.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (wRAR) [reply tests/CrawlerRunner/ip_address.py:33]: If the tests pass, this is the last blocker from me.

**Comment by wRAR** [comment]
flake8:

```
docs/conftest.py:15:1: E302 expected 2 blank lines, found 1
docs/utils/linkfix.py:34:9: F821 undefined name 'sys'
docs/_ext/scrapydocs.py:7:1: E302 expected 2 blank lines, found 1
scrapy/core/scraper.py:281:25: E127 continuation line over-indented for visual indent
scrapy/downloadermiddlewares/stats.py:6:1: E302 expected 2 blank lines, found 1
scrapy/downloadermiddlewares/redirect.py:60:21: E128 continuation line under-indented for visual indent
tests/test_downloadermiddleware_decompression.py:10:1: E303 too many blank lines (3)
tests/CrawlerProcess/reactor_default.py:7:1: E303 too many blank lines (3)
tests/CrawlerProcess/reactor_default_twisted_reactor_select.py:7:1: E303 too many blank lines (3)
```

**Comment by wRAR** [comment]
Also, as we rolled back the changes, these message classes should be put back to pylintrc:

```
************* Module scrapy.http.request
scrapy/http/request/__init__.py:188:11: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck)
************* Module tests.test_request_cb_kwargs
tests/test_request_cb_kwargs.py:94:31: C1803: 'kwargs == {}' can be simplified to 'not kwargs' as an empty sequence is falsey (use-implicit-booleaness-not-comparison)
```

**Comment by marksmayo** [comment]
Apologies, it's been a busy few weeks. Looking now

**Comment by marksmayo** [comment]
@Gallaecio are you able to kick off the workflows, to see if any issues are left after I've resynched and fixed? Thanks.

**Review by wRAR — APPROVED** [review]

**Comment by marksmayo** [comment]
Thanks for your help! Was a bit of a big one to bite off for my first go, in hindsight

File: scrapy/core/downloader/webclient.py [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import re
 from time import time
 from urllib.parse import urlparse, urlunparse, urldefrag
-
 from twisted.web.http import HTTPClient
+
 from twisted.internet import defer
```
Comment (Gallaecio) [inline scrapy/core/downloader/webclient.py:6]: Some import sorting and grouping changes like this one look wrong to me, but I am OK with merging this as an overall improvement, and in a separate pull request introduce a CI job to check these with isort, and include isort into the pre-commit configuration that we may introduce in https://github.com/scrapy/scrapy/pull/5734.

File: tests/CrawlerRunner/ip_address.py [hunk 7]
```diff
@@ -30,9 +29,10 @@ def start_requests(self):
 
     def parse(self, response):
         netloc = urlparse(response.url).netloc
-        self.logger.info("Host: %s" % netloc.split(":")[0])
-        self.logger.info("Type: %s" % type(response.ip_address))
-        self.logger.info("IP address: %s" % response.ip_address)
+        host = netloc.split(":")[0]
+        self.logger.info(f"Host: {host}")
```
Comment (Gallaecio) [inline tests/CrawlerRunner/ip_address.py:33]: Just a tip for the future: using single quotes (`':'`) would have worked as well to move `netloc.split(":")[0]` into the f-string.

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
Thanks!

**Pull request merged**